### PR TITLE
Use gometalinter for linting

### PR DESCRIPTION
**Pull request by codingllama**
Use gometalinter for our linting needs, instead of individually managing a bunch of individual tools.

The metalinter config matches what `presubmit.sh` used to do, with a few additions (gofmt and goimports checks). There's more we could do, but that seemed sufficient for an initial PR.

I've made a few simplifications to `presubmit.sh` and updated `README.md` to reflect our new tooling.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -43,17 +43,15 @@ install:
...
-  - go get github.com/golang/lint/golint
   - go get github.com/golang/protobuf/proto
   - go get github.com/golang/protobuf/protoc-gen-go
-  - go get github.com/kisielk/errcheck
   - go get golang.org/x/tools/cmd/stringer
   - go get github.com/google/certificate-transparency-go
   - go get -d -t github.com/google/certificate-transparency-go/...
   - go install github.com/golang/{mock/mockgen,protobuf/protoc-gen-go}
+  # gometalinter
+  - go get github.com/alecthomas/gometalinter
```
Comment (gdbelvin) [inline .travis.yml]: Do we want to version this or pull from HEAD?  Either way is fine.

Comment (codingllama) [reply .travis.yml]: As long as it doesn't break frequently, pulling from HEAD is probably best, as we don't have to keep manually updating it.

**Comment by codingllama** [comment]
Alright, we're on the green now, ready for review.

@daviddrysdale and @gdbelvin, would you mind having a look?

File: README.md [hunk 2]
```diff
@@ -75,7 +75,7 @@ go get -t -u -v ./...
 To build and run tests, use:
 
 ```bash
-./scripts/presubmit.sh --no-generate --no-linters
+go test ./...
```
Comment (daviddrysdale) [inline README.md]: Yay!

File: README.md [hunk 3]
```diff
@@ -201,12 +201,14 @@ The [`scripts/presubmit.sh`](scripts/presubmit.sh) script runs various tools
...
-go get -u github.com/fzipp/gocyclo
-go get -u github.com/gordonklaus/ineffassign
-go get -u github.com/golang/lint/golint
-# Run checks, tests and code generation
+# Install gometalinter and all linters
+go get -u github.com/alecthomas/gometalinter
+gometalinter --install
+
+# Run linters
+gometalinter --config=gometalinter.json ./...
```
Comment (daviddrysdale) [inline README.md]: If someone follows these instructions as-is, are they going to run the linters twice?

If so, could move this down below the `presubmit` invocation with a comment:
```
# Or just run the linters alone:
gometalinter --config=gometalinter.json ./...
```

File: .travis.yml [hunk 4]
```diff
@@ -43,17 +43,15 @@ install:
...
-  - go get github.com/gordonklaus/ineffassign
-  - go get github.com/golang/lint/golint
   - go get github.com/golang/protobuf/proto
   - go get github.com/golang/protobuf/protoc-gen-go
-  - go get github.com/kisielk/errcheck
   - go get golang.org/x/tools/cmd/stringer
   - go get github.com/google/certificate-transparency-go
   - go get -d -t github.com/google/certificate-transparency-go/...
   - go install github.com/golang/{mock/mockgen,protobuf/protoc-gen-go}
+  # gometalinter
```
Comment (daviddrysdale) [inline .travis.yml]: (nit) comment doesn't add anything of use/interest

File: scripts/presubmit.sh [hunk 5]
```diff
@@ -71,21 +67,19 @@ main() {
...
-    grep -v .pb.go | \
-    grep -v .pb.gw.go | \
-    grep -v _string.go | \
-    grep -v vendor/ | \
-    tr '\n' ' ')"
-  local proto_srcs="$(find . -name '*.proto' | \
-    grep -v vendor/ | \
-    tr '\n' ' ')"
+  cd "$(dirname "$0")"  # at scripts/
+  cd ..  # at /
```
Comment (daviddrysdale) [inline scripts/presubmit.sh]: "at /" is wrong/misleading -- maybe "at top-level"

File: scripts/presubmit.sh [hunk 6]
```diff
@@ -12,35 +12,31 @@
...
   if ! type -p "${cmd}" > /dev/null; then
-    echo "${cmd} not found, try to 'go get -u ${repo}'"
+    echo "${cmd} not found, ${msg}"
     return 1
   fi
 }
 
 usage() {
-  echo "$0 [--coverage] [--fix] [--no-build] [--no-linters] [--no-generate]"
+  echo "$0 [--coverage] [--fix] [--no-build] [--no-lint] [--no-generate]"
```
Comment (daviddrysdale) [inline scripts/presubmit.sh]: The flag-rename seems gratuitous.

File: .travis.yml (hunk 4 above)
Comment (codingllama) [reply .travis.yml]: Removed.

File: README.md (hunk 3 above)
Comment (codingllama) [reply README.md]: Done.

File: scripts/presubmit.sh (hunk 5 above)
Comment (codingllama) [reply scripts/presubmit.sh]: Done.

**Comment by codingllama** [comment]
Thanks for the quick review!

File: scripts/presubmit.sh (hunk 6 above)
Comment (daviddrysdale) [reply scripts/presubmit.sh]: Hardly a big deal, but all my `.bash_history` examples would break...

**Comment by codingllama** [comment]
Btw, fixes #869.

**Pull request merged**